hadoop-common-project/hadoop-common/src/main/java/io/hops/security/ServiceJWTManager.java

```diff
@@ -0,0 +1,592 @@
+            LOG.debug(getName() + " But refreshing JWTs from ssl-server");
+            reloadJWTs();
+            TimeUnit.SECONDS.sleep(sleepPeriodSeconds);
+          } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+          } catch (GeneralSecurityException ex) {
+            LOG.warn("Could not refresh service JWTs when failed to acquire exclusive lock");
+          }
+        }
+      }
+      
+      // Now we have the exclusive lock
+      boolean shouldTry2renew = true;
+      while (!Thread.currentThread().isInterrupted() && shouldTry2renew) {
+        try {
+          // Check if it is time to renew
+          LocalDateTime now = DateUtils.getNow();
+          if (isTime2Renew(now, masterTokenExpiration)) {
+            backOff.reset();
+            LocalDateTime expiresAt = DateUtils.getNow().plus(serviceJWTValidityPeriodSeconds, ChronoUnit.SECONDS);
+            int renewalTokenIdx = 0;
+            while (renewalTokenIdx < renewalTokens.length) {
+              try {
+                // Use one-time token to authenticate
+                ServiceTokenDTO renewedTokens = renewServiceJWT(masterToken.get(), renewalTokens[renewalTokenIdx],
+                    expiresAt, now);
+                String oldMasterToken = masterToken.get();
+                masterToken.set(renewedTokens.jwt.token);
+                masterTokenExpiration = DateUtils.date2LocalDateTime(renewedTokens.jwt.expiresAt);
+                renewalTokens = renewedTokens.renewTokens;
+                try {
+                  // Since renewal of master JWT has gone through, invalidate the old one
+                  invalidateServiceJWT(oldMasterToken);
+                } catch (Exception ex) {
+                  // Do not retry if we failed to invalidate old master token
+                  LOG.warn("Failed to invalidate old service master JWT. Continue...");
+                }
+                sslConf.set(JWT_MANAGER_MASTER_TOKEN_KEY, masterToken.get());
+                for (int i = 0; i < renewalTokens.length; i++) {
+                  String confKey = String.format(JWT_MANAGER_RENEW_TOKEN_PATTERN, i);
+                  sslConf.set(confKey, renewalTokens[i]);
+                }
+                URL sslServerURL = sslConf.getResource(getConfig().get(SSLFactory.SSL_SERVER_CONF_KEY, "ssl-server.xml"));
+                File sslServerFile = new File(sslServerURL.getFile());
+                try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(sslServerFile))) {
+                  sslConf.writeXml(bos);
+                  bos.flush();
+                }
+                LOG.info("Updated service JWT");
+                break;
+              } catch (URISyntaxException ex) {
+                LOG.error("There is an error in service JWT renewal URI: " + serviceJWTRenewPath.toString(), ex);
+                shouldTry2renew = false;
+                break;
+              } catch (Exception ex) {
```

Comment:
Do we really want to catch all exceptions, even interupt exceptions?

Reply:
There is nothing throwing an `InterruptedException` in this block.